Lesson 3

thetribalme opened the pull request.

thetribalme committed:
summary about Lesson 3 added

thetribalme committed:
branch architecture set

thetribalme committed:
tasks conditions added

thetribalme committed:
tasks is done (3/3)

thetribalme committed:
tasks is done (1/3)

thetribalme committed:
added comment to num_translate_adv (1/3)

thetribalme committed:
optimized with .get() (1/3)

thetribalme committed:
task is done (2/3)

IrinaMilekhina commented on Lesson_3/Lesson_3.1-2.py:
```diff
@@ -0,0 +1,84 @@
+print('FIRST PART'.center(100, '-'), '\n')
+
+
+def num_translate(number):
```
что возвращает функция?

IrinaMilekhina commented on Lesson_3/Lesson_3.1-2.py:
```diff
@@ -0,0 +1,84 @@
+    :param number: number you want to translate. ONLY word form
+    :return: printed translation
+    """
+    translations = {
```
почему ты решила размещать словарь внутри функции?

IrinaMilekhina commented on Lesson_3/Lesson_3.3-4.py:
```diff
@@ -0,0 +1,82 @@
+}
+}
+'''
+# task 3
```
почему тут закомментированный код?

IrinaMilekhina commented on Lesson_3/Lesson_3.3-4.py:
```diff
@@ -0,0 +1,82 @@
+#             result[key] = []
+#             result[key].append(name)
+#         else:
+#             result[key].append(name)
```
DRY

IrinaMilekhina commented on Lesson_3/Lesson_3.3-4.py:
```diff
@@ -0,0 +1,82 @@
+    list of names and surnames as a value) as a value"""
+    result = {}
+    for name_surname in personal_data:
+        name = name_surname.split(' ')[0]
```
нужен ли параметр в split?

IrinaMilekhina replied:
name, surname = val.split()

IrinaMilekhina replied:
name, surname = val.split()

IrinaMilekhina commented on Lesson_3/Lesson_3.3-4.py:
```diff
@@ -0,0 +1,82 @@
+        second_key = name[0].capitalize()
+        if first_key not in result:
+            result[first_key] = {}
+            name_based = result[first_key]
```
DRY

IrinaMilekhina commented on Lesson_3/Lesson_3.5.py:
```diff
@@ -0,0 +1,80 @@
+                    noun = choice(nouns)
+                jokes_element.append(noun)
+                used_words.append(noun)
+                adverb = choice(adverbs)
```
очень много переменных в которых легко запутаться
давай придумаем другой путь

thetribalme committed:
optimized code to fewer lines (1/3)
renamed variable with translation dict and made it global
used return instead print as function outcome

thetribalme committed:
optimized code to fewer lines (2/3)
renamed variables

thetribalme committed:
optimized code to fewer lines (2/3)
removed and renamed variables
changed Print to return

thetribalme committed:
optimized code to fewer lines (2/3)
removed and renamed variables
changed Print to return

thetribalme committed:
list changing error fixed (3/3)

thetribalme committed:
changed tasks description to list of issued material
